sdk/keyvault/Azure.Security.KeyVault.Secrets/samples/Sample1_HelloWorld.cs

```diff
@@ -0,0 +1,89 @@
+    /// </summary>
+    [Category("Live")]
+    public partial class HelloWorld
+    {
+        [Test]
+        public void HelloWorldSync()
+        {
+            // Environment variable with the Key Vault endpoint.
+            string keyVaultUrl = Environment.GetEnvironmentVariable("AZURE_KEYVAULT_URL");
+
+            // Instantiate a secret client that will be used to call the service. Notice that the client is using default Azure
+            // credentials. To make default credentials work, ensure that environment variables 'AZURE_CLIENT_ID',
+            // 'AZURE_CLIENT_KEY' and 'AZURE_TENANT_ID' are set with the service principal credentials.
+            var client = new SecretClient(new Uri(keyVaultUrl), new DefaultAzureCredential());
+
+            // Let's create a secret holding bank account credentials valid for 1 year. if the secret
+            // already exists in the key vault, then a new version of the secret is created.
+            string secretName = $"BankAccountPassword-{Guid.NewGuid()}";
+
+            var secret = new Secret(secretName, "f4G34fMh8v")
+            {
+                Expires = DateTimeOffset.Now.AddYears(1)
+            };
+
+            client.Set(secret);
+
+            // Let's Get the bank secret from the key vault.
+            Secret bankSecret = client.Get(secretName);
+            Debug.WriteLine($"Secret is returned with name {bankSecret.Name} and value {bankSecret.Value}");
+
+            // After one year, the bank account is still active, we need to update the expiry time of the secret.
+            // The update method can be used to update the expiry attribute of the secret. It cannot be used to update
+            // the value of the secret.
+            bankSecret.Expires = bankSecret.Expires.Value.AddYears(1);
+            SecretBase updatedSecret = client.Update(bankSecret);
+            Debug.WriteLine($"Secret's updated expiry time is {updatedSecret.Expires}");
+
+            // Bank forced a password update for security purposes. Let's change the value of the secret in the key vault.
+            // To achieve this, we need to create a new version of the secret in the key vault. The update operation cannot
+            // change the value of the secret.
+            var secretNewValue = new Secret(secretName, "bhjd4DDgsa");
+            secretNewValue.Expires = DateTimeOffset.Now.AddYears(1);
+
+            client.Set(secretNewValue);
+
+            // The bank account was closed. You need to delete its credentials from the key vault.
+            client.Delete(secretName);
+
+            // To ensure secret is deleted on server side.
+            Assert.IsTrue(WaitForDeletedKey(client, secretName));
+
+            // If the keyvault is soft-delete enabled, then for permanent deletion, deleted secret needs to be purged.
+            client.PurgeDeleted(secretName);
+
+        }
+
+        private bool WaitForDeletedKey(SecretClient client, string secretName)
```

Comment:
Is this the pattern we want for polling? 

I assume we are polling 20 times because we don't know how long the server will take?

Reply:
So far is what we've been using in the tests so I wanted to show it here too.
As for the number of times, yes, it is just an estimate were we are hoping for the best. I've seen it retry max 7 times, but that is just during my tests so I don't feel comfortable putting 7 :|